pyomo/contrib/pynumero/interfaces/pyomo_grey_box_nlp.py

```diff
@@ -100,8 +117,15 @@ def __init__(self, pyomo_model):
                     fixed_vars.extend(v for v in data.inputs.values() if v.fixed)
                     fixed_vars.extend(v for v in data.outputs.values() if v.fixed)
                     greybox_nlp = _ExternalGreyBoxAsNLP(data)
+                    if greybox_nlp._ex_model.has_objective():
```

Comment:
Should this be `data.get_external_model().has_objective()`?

Reply:
absolutely.

Reply:
done